api/src/shipit_api/admin/github.py

```
@@ -214,19 +221,23 @@ def list_xpis(owner, repo, revision):
         # convert "master" into a stable ref
         ref = xpi.get("branch", config["taskgraph"]["repositories"][xpi["repo-prefix"]]["default-ref"])
         commit = ref_to_commit(xpi_owner, xpi_repo, ref)
-        package = get_package_json(xpi_owner, xpi_repo, commit, directory=xpi.get("directory"))
-        xpis.append(
-            {
-                "revision": commit,
-                "branch": xpi.get("branch", "master"),
-                "version": package["version"],
-                "xpi_name": xpi["name"],
-                "owner": xpi_owner,
-                "repo": xpi_repo,
-                "manifest_revision": revision,
-                "addon-type": xpi["addon-type"],
-            }
-        )
+        try:
+            package = get_package_json(xpi_owner, xpi_repo, commit, directory=xpi.get("directory"))
+            xpis.append(
+                {
+                    "revision": commit,
+                    "branch": xpi.get("branch", "master"),
+                    "version": package["version"],
+                    "xpi_name": xpi["name"],
+                    "owner": xpi_owner,
+                    "repo": xpi_repo,
+                    "manifest_revision": revision,
+                    "directory": xpi.get("directory", ""),
+                    "addon-type": xpi["addon-type"],
+                }
+            )
+        except TypeError:
```

Why are we eating this exception? (This deserves a comment if there's a justifiable reason for it)

---

Maybe we should just send it to Sentry and continue eating it. We do this in Balrog for some situations, it should just be a called to `capture_exception` like in https://github.com/mozilla-releng/balrog/blob/main/src/auslib/web/public/base.py#L102.

I can't think of a great way to surface in the UI either, and it would probably just get in the way.